src/apps/TxBuilder/hooks/useServices/interfaceRepository.ts

```diff
@@ -48,7 +48,13 @@ class InterfaceRepository {
     const abi = JSON.parse(abiString);
 
     const methods = abi
-      .filter((e: any) => e.constant === false)
+      .filter((e: any) => {
+        if (["pure", "view"].includes(e.stateMutability)) {
+          return false;
+        }
+
+        return !!e.constant === false;
```

Comment:
Was !e.constant not enough?

Reply:
nop, we need to filter then the value **is** false

Reply:
Sorry, what?

Reply:
`e.constant` is of type `boolean | undefined` and we can use the function in the ui if `e.constant` is `false` or `undefined`. If it is `undefined` the `!!` will convert it to `false` and for `false` it will stay `false`. Therefore the `!!` is required.

Reply:
this condition runs inside a filter, so, the current item needs to be included in the result when `e.constant` is false or undefined.

!!e.constant === false => true is different to !!e.constant if e.constant is false.

https://solidity.readthedocs.io/en/v0.5.3/abi-spec.html#json

Reply:
I think what mikhail meant is that `!e.constant` actually evaluated to the same. if `e.constant` is `undefined` then `!e.constant` is `true` and if `e.constant` is `false` then `!e.constant` is also true.

Reply:
acain this `undefined` stuff is dark js magic and I don't like it, so I am not making any promises that what I am writing is correct :P

Reply:
That's exactly what I meant Richard, `!` will convert the value to its Boolean equivalent and perform a logical NOT operation

`!!` is used to convert a value to its Boolean equivalent, it does the same thing as `!` first, and then performs one more logical NOT operation on a value
https://link.medium.com/hSsK8ddDs9

Since we then compare the received value to false, `!` seems to be enough and IMO opinion it's better for understanding and readability (as simple as a developer has to follow one Boolean operation instead of 3)

Reply:
Sorry Nico I had to leave this here
![image](https://user-images.githubusercontent.com/16622558/92030979-e5791e80-ed78-11ea-89d0-6fc91a03b3fe.png)

Reply:
changed!